### PR TITLE
PHPC-2440: Remove deprecated Query constructor options

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-2440

Removes "partial", "maxScan", "modifiers", "oplogReplay", and "snapshot" options.

Removes negative "limit" implying true for "singleBatch".

### CODE REVIEW

File: tests/standalone/query-errors.phpt
Comment (jmikola) [inline tests/standalone/query-errors.phpt:2]: `query-ctor_error-002.phpt` makes this test redundant, as it covers these types and more.

File: tests/server/server-executeQuery-007.phpt
Comment (jmikola) [inline tests/server/server-executeQuery-007.phpt:2]: Decided to repurpose this test instead of delete it, as we didn't have any other PHPC tests for `singleBatch: true`.

File: tests/query/query-ctor-005.phpt
Comment (jmikola) [inline tests/query/query-ctor-005.phpt:7]: As mentioned in [this comment in PHPC-2440](https://jira.mongodb.org/browse/PHPC-2440?focusedCommentId=6733311&focusedId=6733311&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-6733311), the server will raise an error. I decided to leave this test in place just to demonstrate that we no longer have validation.

I wasn't interested in adding a functional test for the server behavior.

File: tests/query/query-ctor-002.phpt
Comment (jmikola) [inline tests/query/query-ctor-002.phpt:75]: I removed `php_phongo_query_init_limit_and_singlebatch()`, so these options are now parsed alphabetically alongside other options. Since those parsing macros are responsible for appending them to the cursor's BSON options, it impacted the debug output order (where we dump the options document as-is).

File: src/MongoDB/Query.c
Comment (alcaeus) [inline src/MongoDB/Query.c:209]: Noted that we'll add a deprecation for this in PHPC-2464.

**Review by alcaeus — APPROVED** [review]

File: UPGRADE-2.0.md
Comment (jmikola) [inline UPGRADE-2.0.md:40]: @alcaeus: I realized I forgot to add upgrade notes before merging. Doing so now and will merge this down once CI comes back green.

**Pull request merged**
